tracer/test/Datadog.Trace.Tests/SpanTests.cs

```diff
@@ -360,5 +360,45 @@ public void SetTag_Double()
             span.GetTag(stringKey).Should().Be(stringValue);
             span.GetTag(numericKey).Should().BeNull();
         }
+
+        [Fact]
+        public void ServiceOverride_WhenSet_HasBaseService()
+        {
+            var origName = "MyServiceA";
+            var span = _tracer.StartSpan(nameof(SetTag_Double), serviceName: origName);
+            span.ServiceName = "MyServiceB";
+
+            span.GetTag(Tags.BaseService).Should().Be(origName);
+        }
+
+        [Fact]
+        public void ServiceOverride_WhenNotSet_HasNoBaseService()
+        {
+            var origName = "MyServiceA";
+            var span = _tracer.StartSpan(nameof(SetTag_Double), serviceName: origName);
+
+            span.GetTag(Tags.BaseService).Should().BeNull();
+        }
+
+        [Fact]
+        public void ServiceOverride_WhenSetSame_HasNoBaseService()
+        {
+            var origName = "MyServiceA";
+            var span = _tracer.StartSpan(nameof(SetTag_Double), serviceName: origName);
+            span.ServiceName = origName;
+
+            span.GetTag(Tags.BaseService).Should().BeNull();
+        }
+
+        [Fact]
+        public void ServiceOverride_WhenSetTwice_HasBaseService()
+        {
+            var origName = "MyServiceA";
+            var span = _tracer.StartSpan(nameof(SetTag_Double), serviceName: origName);
+            span.ServiceName = "MyServiceB";
+            span.ServiceName = "MyServiceC";
+
+            span.GetTag(Tags.BaseService).Should().Be(origName);
+        }
     }
```

Comment:
Optional extra test (if this is the behavior we want, otherwise we should update the test!)
```suggestion
        [Fact]
        public void ServiceOverride_WhenSetSameWithDifferentCase_HasNoBaseService()
        {
            var origName = "MyServiceA";
            var newName = "MYSERVICEA";
            var span = _tracer.StartSpan(nameof(SetTag_Double), serviceName: origName);
            span.ServiceName = newName;

            span.ServiceName.Should().Be(newName);
            span.GetTag(Tags.BaseService).Should().BeNull();
        }
    }
```